### PR TITLE
#63 get from sql for user

### PR DESCRIPTION
**Pull request by davinci-school**

### CODE REVIEW

**Comment by davinci-school** [comment]
https://skoladavinci.tpondemand.com/entity/63

File: backend/server.js [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+app.get('/users', (req, res) => {
+  const allUsernames = Object.keys(Database);
+  console.log('list of allUsernames is:', allUsernames);
+  res.send(allUsernames);
+});
+
+app.get('/user_data_past_month', redirectLogin, (req, res) => {
+
+
+  connection.query('SELECT * FROM time_board WHERE ID_users = 2 AND date BETWEEN SUBDATE(CURDATE(), INTERVAL 1 MONTH) AND NOW();', [req.session.userid])
```
Comment (davinci-school) [inline backend/server.js]: `ID_users = 2` in sql query should be replaced by `ID_users = ?`

File: backend/server.js [hunk 2]
```diff
@@ -163,9 +165,11 @@ app.get('/users', (req, res) => {
 app.get('/user_data_past_month', redirectLogin, (req, res) => {
 
 
-  connection.query('SELECT * FROM time_board WHERE ID_users = 2 AND date BETWEEN SUBDATE(CURDATE(), INTERVAL 1 MONTH) AND NOW();', [req.session.userid])
+  connection.query('SELECT * FROM time_board WHERE ID_users = ? AND date BETWEEN SUBDATE(CURDATE(), INTERVAL 1 MONTH) AND NOW();', [req.session.userid])
 
-    .then(results => res.send(results))
+    .then(results=JSON.parse(results);)
```
Comment (davinci-school) [inline backend/server.js]: syntax error, extra ';'

Comment (davinci-school) [inline backend/server.js]: syntax error, extra ';'

Comment (OndrejSchicker) [reply backend/server.js]: solved

Comment (OndrejSchicker) [reply backend/server.js]: solved

File: backend/server.js [hunk 3]
```diff
@@ -163,9 +165,11 @@ app.get('/users', (req, res) => {
...
+    .then(console.log(results));
+    .then(results => res.send(results.name, results.time_in, results.time_out))
```
[the remark this answers is not in the document]
Comment (OndrejSchicker) [reply backend/server.js]: Solved by removing extra JSON.parse

**Pull request merged**